src/services/stepsService.ts

```diff
@@ -79,27 +78,27 @@ export class StepsService {
     if (!steps) return null;
 
     return steps.reduce((list: IStepProps[], step: IStepProps) => {
-        let clone: IStepProps | null = null;
+      let clone: IStepProps | null = null;
 
-        if (predicate(step) && steps.some((s) => s.UUID === step.UUID)) {
-          // clone the step if it matches the condition and isn't a nested step
-          clone = Object.assign({}, step);
-        }
+      if (predicate(step) && steps.some((s) => s.UUID === step.UUID)) {
+        // clone the step if it matches the condition and isn't a nested step
+        clone = Object.assign({}, step);
+      }
 
-        // overwrite the branch if one of its steps contains a match
-        if (clone && clone.branches) {
-          clone.branches.forEach((branch, idx) => {
-            const filteredBranchSteps = StepsService.filterNestedSteps(branch.steps, predicate);
-            if (filteredBranchSteps && clone?.branches) {
-              clone.branches[idx].steps = filteredBranchSteps;
-            }
-          });
-        }
+      // overwrite the branch if one of its steps contains a match
+      if (clone && clone.branches) {
+        clone.branches.forEach((branch, idx) => {
+          const filteredBranchSteps = StepsService.filterNestedSteps(branch.steps, predicate);
+          if (filteredBranchSteps && clone?.branches) {
```

Comment:
Ooh good catch, I'll update this, thanks!

Reply:
So, I remembered why I added that check there, it's because the null checking is giving problems:
![Screen Shot 2023-02-13 at 10 17 12 am](https://user-images.githubusercontent.com/3844502/218431551-49e30516-5069-4385-a2e3-ef1494d666e8.png)

Where even if we do a check for that property, it doesn't "remember" that we've already done this check earlier:
![Screen Shot 2023-02-13 at 10 17 34 am](https://user-images.githubusercontent.com/3844502/218431574-d3fd9587-f791-482a-83fd-c350413008f0.png)

So one other way to do this would be to add an exclamation mark to say "we will definitely have this value at runtime" like this:
```
clone!.branches![idx].steps = filteredBranchSteps;
```
Do you prefer this instead of the `if` check?